test/test_encryption.py

```diff
@@ -167,36 +168,48 @@ def test_init_spawn_args(self):
     @unittest.skipUnless(_HAVE_PYMONGOCRYPT, "pymongocrypt is not installed")
     def test_init_kms_tls_options(self):
         # Error cases:
+        opts = AutoEncryptionOpts({}, "k.d", kms_tls_options={"kmip": 1})
         with self.assertRaisesRegex(TypeError, r'kms_tls_options\["kmip"\] must be a dict'):
-            AutoEncryptionOpts({}, "k.d", kms_tls_options={"kmip": 1})
+            client = self.rs_or_single_client(auto_encryption_opts=opts)
+            client.db.coll.insert_one({"encrypted": "test"})
```

Comment:
These tests should raise from the AsyncMongoClient constructor, without even attempting to connect. We could even use AsyncMongoClient() directly:
```python
        with self.assertRaisesRegex(TypeError, r'kms_tls_options\["kmip"\] must be a dict'):
            AsyncMongoClient(auto_encryption_opts=opts)
```

Could you confirm?

Reply:
Ahhh yeah, i see now. Thanks for popping over to further explain it, and for working thru the soln with me!